IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormPushTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CommonTestUtils;
+using FluentAssertions;
+using GitCommands.Git;
+using GitUI;
+using GitUI.CommandsDialogs;
+using NUnit.Framework;
+
+namespace GitExtensions.UITests.CommandsDialogs
+{
+    [Apartment(ApartmentState.STA)]
+    public class FormPushTests
+    {
+        // Created once for the fixture
+        private ReferenceRepository _referenceRepository;
+
+        // Created once for each test
+        private GitUICommands _commands;
+
+        [SetUp]
+        public void SetUp()
+        {
+            if (_referenceRepository == null)
+            {
+                _referenceRepository = new ReferenceRepository();
+            }
+            else
+            {
+                _referenceRepository.Reset();
+            }
+
+            _commands = new GitUICommands(_referenceRepository.Module);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            _referenceRepository.Dispose();
+        }
+
+        [TestCase(false, false, false, ForcePushOptions.DoNotForce)]
+        [TestCase(false, true, true, ForcePushOptions.Force)]
+        [TestCase(false, true, false, ForcePushOptions.DoNotForce)] // ForcePushBranches state is bound to ForcePushTags state
```

Comment:
The state bounding actually depends on whether or not `GitVersion.Current.SupportPushForceWithLease`, so this needs modified in some way.